client/src/components/board-details/board-information/BoardDetails.tsx

```diff
@@ -0,0 +1,125 @@
+"use client";
+
+import React from "react";
+import { Button } from "@/components/ui/button";
+import { Progress } from "@/components/ui/progress";
+import { BoardHeader } from "@/components/user-boards/BoardHeader";
+import BoardInfoInputItem from "./BoardInfoInputItem";
+import CollaboratingUsers from "./CollaboratingUsers";
+import BoardNameField from "./BoardNameField";
+import useSWR from "swr";
+import { fetchBoardDetails } from "@/mock-data/BoardDetailsFetcher";
+import BoardDetailsInfoSkeleton from "@/components/loading/BoardDetailsInfoSkeleton";
+import BoardInfoItem from "./BoardInfoItem";
+import { useRouter } from "next/navigation";
+
+const BoardDetails: React.FC<{ boardId: string }> = ({ boardId }) => {
+  const { data: board, error, isLoading, mutate } = useSWR(`/mocked/boards/${boardId}`, fetchBoardDetails);
+  const router = useRouter();
+  if (isLoading) {
+    return <BoardDetailsInfoSkeleton />;
+  }
+
+  if (error) {
+    return (
+      <section className="rounded-lg border border-error-border bg-card p-6 shadow">
+        <BoardHeader title="Board Details" description="Board Information" />
+        <p className="text-error-foreground">{error.message}</p>
+      </section>
+    );
+  }
+
+  if (!board) {
+    return (
+      <section className="rounded-lg border border-border bg-card p-6 shadow">
+        <BoardHeader title="Board Details" description="Board Information" />
+        <p>No board data.</p>
+      </section>
+    );
+  }
+
+  const creationDate = board.createdAt ? new Date(board.createdAt).toLocaleString() : "";
+  const lastUpdated = board.modifiedAt ? new Date(board.modifiedAt).toLocaleString() : "";
+
+  const totalSizeUsed = board.size;
+  const usedPercentage = board.maxSize ? (totalSizeUsed / board.maxSize) * 100 : 0;
+
+  const collaboratingUsers = [
+    ...(board.permissions.viewer || []),
+    ...(board.permissions.editor || []),
+    ...(board.permissions.moderator || []),
+  ];
```

Comment:
I will do it, but not here. Sorting is a complex operation that also requires the use of `useMemo`. Since I only need to perform it in the `CollaboratingUsers` component, I will create the `useMemo` with sorting there.